Add some SSH notes (#2639)
codenvy/che-docs: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc

* [.underline]#The URLs for starting a new workspace#
* The URLs of your workspaces in use

> With {prod-short}, you can visit a URL in your browser to start a new workspace that contains a clone of a Git repository. You can use either an HTTPS or SSH URL to start a new workspace with a clone of a Git repository.

Reviewer: ```suggestion
With {prod-short}, you can use either an HTTPS or SSH URL to start a new workspace with a clone of a Git repository.
```

[IMPORTANT]
====
//TODO: point to the dashboard SSH procedure when https://github.com/eclipse/che/issues/22111 is done.

> * If an SSH URL is used to start a new workspace, the SSH key must be propagated, see link:https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations[Configuring DevWorkspaces to use SSH keys for Git operations].

Reviewer 2: This point should be stripped or reworked for downstreaming; I don't think we want to point to the upstream github repo for docs in downstream. 

Maybe we can adopt the `Configuring DevWorkspaces to use SSH keys for Git operations` section into Che docs? The file is already in `.adoc` format, though would need a style pass, etc.

Author: I don't think we need to have a dedicated page about configuring SSH by a secret. When https://github.com/eclipse/che/issues/22111 is done we will point to the user friendly dashboard procedure, so the current procedure will be redundant as a too complicated one. @ibuziuk **WDYT?**

Reviewer 2: I agree we don't need a dedicated page and the link to DWO's doc is sufficient here. My concern is that we need to do some additional work when we downstream these docs later, as I don't think we should be linking to Github there (this is not a necessarily concern for this repo, though).

Reviewer: if I understand it correctly, you'll make this part visible later?

Author: @deerskindoll 
> if I understand it correctly, you'll make this part visible later?

This is already visible:
![screenshot-0 0 0 0_4000-2023 08 23-15_51_13](https://github.com/eclipse-che/che-docs/assets/7668752/f9bd318e-9b61-4ce8-bd86-dc5957d82438)
It should be modified later

Reviewer: got it! thanks for the clarification

> * If an SSH URL is used to start a new workspace, the SSH key must be propagated, see link:https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations[Configuring DevWorkspaces to use SSH keys for Git operations].

Reviewer: ```suggestion
* If you use a SSH URL to start a new workspace, you must propagate the SSH key must be propagated. See link:https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations[Configuring DevWorkspaces to use SSH keys for Git operations] for more information.
```

> * If the SSH URL points to a private repository, to be able to fetch the `devfile.yaml` content, an access token must be applied by accepting an SCM authentication page or xref:using-a-git-provider-access-token.adoc[Personal Access Token] procedures.

Reviewer 3: for private repos both PAT and OAuth should work, right?

Author: yes

> * If the SSH URL points to a private repository, to be able to fetch the `devfile.yaml` content, an access token must be applied by accepting an SCM authentication page or xref:using-a-git-provider-access-token.adoc[Personal Access Token] procedures.

Reviewer: ```suggestion
* If the SSH URL points to a private repository, you must apply an access token to be able to fetch the `devfile.yaml` content. You can do this either by accepting an SCM authentication page or following a xref:using-a-git-provider-access-token.adoc[Personal Access Token] procedure.
```
